### PR TITLE
Create user sessions

### PR DESCRIPTION
**Pull request by tchoutri**
## Proposed changes

This PR adds the infrastructure for user sessions.

* [x] Users can log in the system
* [x] Users who are already logged-in are redirected to the homepage if they try to log-in again
* [x] Have a proper drop-down menu to display relevant user/site links & options
  * [x] log-in/username
  * [x] guides
  * [x] dark / light mode
  * [x] log-off when applicable
* [x] Cookies have a TTL
* [x] Users can log-off
---

### References

* https://owasp.org/www-community/HttpOnly
* https://owasp.org/www-pdf-archive//OWASPLondon20171130_Cookie_Security_Myths_Misconceptions_David_Johansson.pdf
* https://owasp.org/www-community/controls/SecureCookieAttribute
* https://developer.mozilla.org/fr/docs/Web/HTTP/Headers/Set-Cookie

### CODE REVIEW

File: src/FloraWeb/Server.hs [hunk 1]
```diff
@@ -60,11 +63,24 @@ runServer floraEnv = withStdoutLogger $ \logger -> do
...
+      hoistServer (Proxy @Pages.Routes)
+                  (withReaderT (const session) Pages.server)
   }
 
+{-
+  si j'ai un cookie, je le décode
+    -> récupère le session_id
+    -> récupère la session depuis la db
+  si j'ai pas de cookie, je créé une session 
+    -> créé un cookie vide de variables de templates et dépose-le
```
Comment (elland) [inline src/FloraWeb/Server.hs:76]: Interesting choice to document in french 😄

Comment (tchoutri) [reply src/FloraWeb/Server.hs:76]: That's become it's still a draft ;-)

**Comment by tchoutri** [comment]
@jappeace I'm getting the following error:

```haskell
 Setup: Encountered missing or private dependencies:
http-client ==0.7.10
```

However, adding it to the ci.nix file gives me this:

```nix
error: infinite recursion encountered

       at /nix/store/hy04g0cmmsvkwabwas03d7kjv1y40pd9-source/pkgs/stdenv/generic/make-derivation.nix:244:7:

          243|       depsBuildBuild              = lib.elemAt (lib.elemAt dependencies 0) 0;
          244|       nativeBuildInputs           = lib.elemAt (lib.elemAt dependencies 0) 1;
             |       ^
          245|       depsBuildTarget             = lib.elemAt (lib.elemAt dependencies 0) 2;
(use '--show-trace' to show detailed location information)
make: *** [Makefile:64: nix-build] Error 1
```

**Comment by jappeace** [comment]
https://github.com/flora-pm/flora-server/pull/61

**Comment by jappeace** [comment]
:tada:

**Pull request merged**